scripts/GasEstimation.ts

```diff
@@ -0,0 +1,435 @@
+} from '@rsksmart/rif-relay-client';
+import { UtilToken } from '@rsksmart/rif-relay-contracts';
+import { BigNumber, ContractReceipt, Wallet, constants } from 'ethers';
+import {
+  createSmartWalletFactory,
+  createSupportedSmartWallet,
+  deployRelayHub,
+  getSuffixDataAndSignature,
+  SupportedSmartWallet,
+} from '../test/utils/TestUtils';
+import { TestRecipient, TestVerifierEverythingAccepted } from 'typechain-types';
+import {
+  SmartWalletFactory,
+  Penalizer,
+  RelayHub,
+  SmartWallet,
+} from '@rsksmart/rif-relay-contracts';
+import { expect } from 'chai';
+
+const GAS_PRICE = '60000000';
+const GAS_LIMIT = 4e6;
+const UNDERLINE = '\x1b[4m';
+const RESET = '\x1b[0m';
+const TOKEN_AMOUNT_TO_TRANSFER = ethers.utils.parseUnits('100').toString();
+
+let relayWorker: SignerWithAddress;
+let relayManager: SignerWithAddress;
+let relayOwner: SignerWithAddress;
+let fundedAccount: SignerWithAddress;
+let relayHubSigner: SignerWithAddress;
+
+function combineTwoRelayRequests(
+  baseRelayRequest: RelayRequest,
+  overrideRelayRequest: {
+    request?: Partial<RelayRequestBody>;
+    relayData?: Partial<EnvelopingRequestData>;
+  }
+) {
+  return {
+    request: { ...baseRelayRequest.request, ...overrideRelayRequest.request },
+    relayData: {
+      ...baseRelayRequest.relayData,
+      ...overrideRelayRequest.relayData,
+    },
+  };
+}
+
+function logTitle(title: string) {
+  console.log(`${UNDERLINE}\n${title.toUpperCase()}${RESET}`);
+}
+
+async function deployContract<Contract>(contract: string) {
+  const contractFactory = await ethers.getContractFactory(contract);
+
+  return contractFactory.deploy() as Contract;
+}
+
+async function setupRelayHub(relayHub: RelayHub) {
+  await relayHub
+    .connect(relayOwner)
+    .stakeForAddress(relayManager.address, 1000, {
+      value: ethers.utils.parseEther('2'),
+    });
+
+  await relayHub.connect(relayManager).addRelayWorkers([relayWorker.address]);
+  await relayHub.connect(relayManager).registerRelayServer('http://relay.com');
+}
+
+async function deployAndSetup() {
+  const owner = Wallet.createRandom().connect(ethers.provider);
+  const penalizer = await deployContract<Penalizer>('Penalizer');
+  const verifier = await deployContract<TestVerifierEverythingAccepted>(
+    'TestVerifierEverythingAccepted'
+  );
+  const token = await deployContract<UtilToken>('UtilToken');
+  const smartWalletTemplate = await deployContract<SmartWallet>('SmartWallet');
+  const relayHub = await deployRelayHub(penalizer.address);
+  await setupRelayHub(relayHub);
+
+  await fundedAccount.sendTransaction({
+    to: owner.address,
+    value: ethers.utils.parseEther('1'),
+  });
+
+  const smartWalletFactory = (await createSmartWalletFactory(
+    smartWalletTemplate,
+    false,
+    owner
+  )) as SmartWalletFactory;
+
+  return {
+    relayHub,
+    owner,
+    smartWalletFactory,
+    token,
+    verifier,
+  };
+}
+
+async function getTransferEstimationWithoutRelay(token: UtilToken) {
```

Comment:
I don't understands the purpose of this function, or maybe we can improve the name.